tests/asgi/tests.py

```diff
@@ -235,11 +235,8 @@ def __call__(self, **kwargs):
         # Give response.close() time to finish.
         await communicator.wait()
 
-        # At this point, AsyncToSync does not have a current executor. Thus
-        # SyncToAsync falls-back to .single_thread_executor.
-        target_thread = next(iter(SyncToAsync.single_thread_executor._threads))
+        # AsyncToSync should have executed the signals in the same thread.
         request_started_thread, request_finished_thread = signal_handler.threads
-        self.assertEqual(request_started_thread, target_thread)
-        self.assertEqual(request_finished_thread, target_thread)
+        self.assertEqual(request_started_thread, request_finished_thread)
```

Comment:
This is probably fine... We loose the **this thread** check but...

Reply:
I tried to backport the necessary part of 36fa071d6ebd18a61c4d7f1b5c9d17106134bd44 to make it works with all supported versions of `asgiref`. We don't have this check in 4.0+ branches :thinking:

Reply:
It's likely fine. 🤔

Let me have a play in the debugger tomorrow.